### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by alex-westwood**
Add unit tests and documentation README file.

### CODE REVIEW

**Review by fryerd1 — COMMENTED** [review]
No errors found 8/8 tests passed, potentially change instructions to be slightly more descriptive.

**Comment by alex-westwood** [comment]
> No errors found 8/8 tests passed, potentially change instructions to be slightly more descriptive.

Thanks Dylan. Is it the README file you're referring to here?

**Comment by fryerd1** [comment]
> > No errors found 8/8 tests passed, potentially change instructions to be slightly more descriptive.
> 
> Thanks Dylan. Is it the README file you're referring to here?

I think that was referring to the RAP unit testing guide.

File: tests/RAP Unit testing guide.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- **Transparency:** Tests document what your code is supposed to do, making it easier to review and maintain.
+- **Quality:** Tests catch bugs and edge cases before code is used in production or shared with others.
+
+## How unit tests work in RAP
+- Each module (e.g., io, cleaning, processing) has its own test file in the `tests/` folder.
+- Each function has one or more tests to ensure that it produces an expected output for different scenarios.
+- If any test fails, the developer must review the error message, fix the code, and re-run the tests.
+
+## How to run unit tests
+- To run all unit tests, open a terminal in the project root and enter:
```
Comment (alex-westwood) [inline tests/RAP Unit testing guide.md:25]: Add if you have not already installed dependencies

pip install -r requirements.txt

**Review by fryerd1 — APPROVED** [review]
